app/gui/src/controller/searcher.rs

```diff
@@ -482,6 +482,53 @@ impl Data {
     }
 }
 
+/// A helper wrapper for the state needed to provide the list of visible components.
+#[derive(Clone, Debug, CloneRef)]
+pub struct ComponentsProvider {
+    breadcrumbs:  Breadcrumbs,
+    list:         component::List,
+    has_this_arg: Rc<bool>,
+}
```

Comment:
1. Why components provider needs access to breadcrumbs?
2. There is no explanation of what is "list" and what is "has_this_arg".